### ISSUE TITLE
Tab completion should not replace ~ or variables

### ISSUE THREAD

**Issue by lzybkr** [PowerShell/PowerShell#5350]
Tab completion should not replace ~ or variables.

E.g.

```powershell
cd ~\<TAB>
vim $env:APPDATA\<TAB>
& $env:SystemRoot\<TAB>
```

When editing a script, the replacement might actually make the script less portable, e.g. `$env:SystemRoot` isn't always on the `c:` drive.

On the command line, you normally end up with a longer command line, and possibly less meaningful. It can also be less useful when searching your history.

We could have an option that control this behavior, but I suspect it should be the default.

**Comment by iSazonov** [comment]
Can we have a way (Alt-Shift) to replace ~ or variables still?

**Comment by lzybkr** [comment]
A key binding depends on the host - e.g. PSReadLine or PS Editor Services - not something the core engine needs to provide, except perhaps an api to do the replacement.

A new api probably isn't worth it, various editors also want to support older versions of PowerShell and this replacement isn't that hard to implement with existing apis.

**Comment by masterflitzer** [comment]
yes this is so annoying, especially when working with both bash and pwsh on a regular basis (i notice it every single day)
